include/boost/math/tools/bivariate_statistics.hpp

```diff
@@ -80,7 +81,7 @@ auto correlation_coefficient(Container const & u, Container const & v)
     // If one dataset is constant and the other isn't, then they have no correlation:
     if (Qu == 0 || Qv == 0)
     {
-        return Real(0);
+        return std::numeric_limits<Real>::quiet_NaN();
```

Comment:
I think lines 76-79L should be deleted also. If both datasets are constant, then the correlation is still undefined.

Reply:
Have you worked out the correlation_coefficient(x,y) when x and y are constant vectors contaminated by additive Gaussian white noise in the limit of the noise variance decaying to zero? I think this is a relevant calculation . . .

Reply:
I think that's 0, both empirically and intuitively, right?

```Mathematica
x = Table[
   {
    N@var,
    Correlation[
     RandomVariate[NormalDistribution[0, var], 100000],
     RandomVariate[NormalDistribution[0, var], 100000]
     ]
    },
   {var, 10^Range[0, -10, -1]}
   ];
x // TableForm
```
```
1.           0.000770873
0.1          0.00157435
0.01        -0.00303611
0.001        0.00186714
0.0001      -0.0000911521
0.00001      0.000377327
1.*10^-6    -0.0000343936
1.*10^-7    -0.00773857
1.*10^-8     0.000511923
1.*10^-9     0.000770162
1.*10^-10   -0.00422592
```
```
Mean@x[[All,2]]
-0.000841306
```

Are you advocating for returning 0 on this basis though?

Reply:
> Are you advocating for returning 0 on this basis though?

Yes.

Reply:
Actually I've come around to your point of view on this. Looking at the [wiki](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) there are tons of different ways to approach that limit, and I really don't see why I thought that additive white Gaussian noise was the "canonical" limit of interest.